mlflow/models/resources.py

```diff
@@ -108,21 +121,16 @@ class DatabricksServingEndpoint(DatabricksResource):
 
     Args:
         endpoint_name (str): The name of all the databricks endpoints used by the model.
+        on_behalf_of_user (Optional[bool]): Flag indicating if the resource
+        is used on behalf of a user.
```

Comment:
```suggestion
        on_behalf_of_user (Optional[bool]): If True, the serving endpoint is accessed with
            with the permission of the invoker of the model. Otherwise ....
```

1. Can we add more specific description? (also for other resources)
2. If we make this optional, we should describe the behavior when it is set to `None`.